### PR TITLE
[storage] [lint] Re-enable several rules

### PR DESCRIPTION
**Pull request by glimchb**
- [storage] [lint] Re-enable rule=core::0135
- [storage] [lint] Re-enable rule=core::0131
- [storage] [lint] Re-enable rule=core::0132
- [storage] [lint] Re-enable rule=core::0134
- [storage] [lint] Re-enable rule=core::0133

after this PR:
```
+--------------------------------------+------------------+----------------+
|                 RULE                 | TOTAL VIOLATIONS | VIOLATED FILES |
+--------------------------------------+------------------+----------------+
| core::0156::forbidden-methods        |               11 |              7 |
| core::0127::http-annotation          |               12 |              7 |
| core::0127::resource-name-extraction |               22 |              7 |
| core::0192::has-comments             |              380 |              8 |
+--------------------------------------+------------------+----------------+
```

### CODE REVIEW

File: storage/Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ all:
...
 	docker run --user=$$(id -u):$$(id -g) --rm --entrypoint=sh -v "${PWD}/../common/v1":/common -v "${PWD}"/v1alpha1/:/out -w /out -v "${PWD}"/v1alpha1:/protos pseudomuto/protoc-gen-doc -c "protoc -I /common -I /protos --doc_out=/out --doc_opt=markdown,autogen.md /protos/*.proto /common/*.proto"
+	docker run --user=$$(id -u):$$(id -g) --rm --entrypoint=sh -v "${PWD}/../common/v1":/common -v "${PWD}"/v1alpha1/:/out -w /out ghcr.io/docker-multiarch/google-api-linter:1.36.0 -c "api-linter -I /common /out/*.proto --output-format summary"
+	docker run --user=$$(id -u):$$(id -g) --rm --entrypoint=sh -v "${PWD}/../common/v1":/common -v "${PWD}"/v1alpha1/:/out -w /out ghcr.io/docker-multiarch/google-api-linter:1.36.0 -c "api-linter -I /common /out/*.proto --output-format github --disable-rule=core::0192::has-comments --disable-rule=core::0127 --disable-rule=core::0131 --disable-rule=core::0132 --disable-rule=core::0133 --disable-rule=core::0134  --disable-rule=core::0156 --set-exit-status"
```
Comment (seroyer) [inline storage/Makefile]: Was this a typo?  rule 156 vs 136?

File: storage/v1alpha1/backend_null.proto [hunk 2]
```diff
@@ -10,39 +10,49 @@ option java_outer_classname = "BackendNullProto";
...
 import "object_key.proto";
 import "uuid.proto";
 
 // Back End (network-facing) APIs. This is debug interface for null block devices.
 service NullDebugService {
     rpc CreateNullDebug (CreateNullDebugRequest) returns (NullDebug) {
         option (google.api.http) = {
-            post: "/v1/subsystems"
-            body: "subsystem"
+            post: "/v1/{parent=subsystems}"
```
Comment (glimchb) [inline storage/v1alpha1/backend_null.proto:27]: don't review URLs, they will be fixed here #188

File: storage/v1alpha1/backend_nvme_tcp.proto [hunk 3]
```diff
@@ -10,38 +10,48 @@ option java_outer_classname = "BackendNvmeTcpProto";
...
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
+
 import "object_key.proto";
 
 // Back End (network-facing) APIs. NVMe/TCP and NVMe/RoCEv2 protocols are covered by this service.
 service NVMfRemoteControllerService {
     rpc CreateNVMfRemoteController (CreateNVMfRemoteControllerRequest) returns (NVMfRemoteController) {
         option (google.api.http) = {
-            post: "/v1/subsystems"
```
Comment (glimchb) [inline storage/v1alpha1/backend_nvme_tcp.proto:21]: don't review URLs, they will be fixed here #188

File: storage/v1alpha1/frontend_virtio_scsi.proto [hunk 4]
```diff
@@ -11,95 +11,115 @@ option java_outer_classname = "FrontendVirtioScsiProto";
...
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
 
 // Front End (host-facing) APIs. Mostly used for Virtio-scsi emulation and host presentation as alternative to Virtio-blk.
 service FrontendVirtioScsiService {
     rpc CreateVirtioScsiTarget (CreateVirtioScsiTargetRequest) returns (VirtioScsiTarget) {
         option (google.api.http) = {
-            post: "/v1/virtioscsitargets"
-            body: "virtioscsitarget"
+            post: "/v1/{parent=subsystems}"
```
Comment (glimchb) [inline storage/v1alpha1/frontend_virtio_scsi.proto:26]: don't review URLs, they will be fixed here #188

File: storage/v1alpha1/frontend_virtio_blk.proto [hunk 5]
```diff
@@ -11,37 +11,47 @@ option java_outer_classname = "FrontendVirtioBlkProto";
...
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
 
 // Front End (host-facing) APIs. Mostly used for Virtio-blk emulation emulation and host presentation as alternative to Nvme.
 service FrontendVirtioBlkService {
     rpc CreateVirtioBlk (CreateVirtioBlkRequest) returns (VirtioBlk) {
         option (google.api.http) = {
-            post: "/v1/virtioblks"
-            body: "virtioblk"
+            post: "/v1/{parent=subsystems}"
```
Comment (glimchb) [inline storage/v1alpha1/frontend_virtio_blk.proto:26]: don't review URLs, they will be fixed here #188

File: storage/v1alpha1/frontend_nvme_pcie.proto [hunk 6]
```diff
@@ -13,95 +13,115 @@ option go_package = "github.com/opiproject/opi-api/storage/v1alpha1/gen/go";
...
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
 
 // Front End (host-facing) APIs. Mostly used for NVMe/PCIe emulation and host presentation.
 service FrontendNvmeService {
     rpc CreateNVMeSubsystem (CreateNVMeSubsystemRequest) returns (NVMeSubsystem) {
         option (google.api.http) = {
-            post: "/v1/subsystems"
-            body: "subsystem"
+            post: "/v1/{parent=subsystems}"
```
Comment (glimchb) [inline storage/v1alpha1/frontend_nvme_pcie.proto:28]: don't review URLs, they will be fixed here #188

File: storage/Makefile (hunk 1 above)
Comment (glimchb) [reply storage/Makefile]: no type, this is correct

**Review by seroyer — APPROVED** [review]

**Review by mestery — APPROVED** [review]

**Comment by llabordehpe** [comment]
What are these new "parent" fields in the CreateXxx messages ? What is the API client supposed to put in this field ?
And what are these new <object>_id fields in the CreateXxx messages ? Is it a new caller's handle ? How does it relates to the key in the <object>Spec message?
And why this "nv_me" ?

**Comment by glimchb** [comment]
good questions @llabordehpe 
the answers are https://google.aip.dev/132

**Pull request merged**